test/parse.jl

```diff
@@ -366,8 +366,14 @@ let a⊂b = reduce(&, x ∈ b for x in a) && length(b)>length(a)
     @test !([1,2] ⊂ [1,2])
 end
 
+
 # issue #9503
 @test parse("x<:y") == Expr(:(<:), :x, :y)
 @test parse("x>:y") == Expr(:(>:), :x, :y)
 @test parse("x<:y<:z").head === :comparison
 @test parse("x>:y<:z").head === :comparison
+
+# issue #11169
+@test_throws UndefVarError undefvar
+let undefvar = return end
```

Comment:
I think this should succeed:
```Julia
uncalled(x) = @test false
fret() = uncalled(return true; @test false)
@test fret()
```
Alas it doesn't; the `@test false` after the `return true` is executed. This looks like a bug to me.